platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -639,3 +641,294 @@ void SwitchStateBase::send_fdb_event_notification(
+    }
+
+    auto &all_bridge_ports = m_objectHash.at(SAI_OBJECT_TYPE_BRIDGE_PORT);
+    bool port_id_found = false;
+    const char *hwifname = nullptr;
+    for (auto& bp: all_bridge_ports)
+    {
+        if (bp.first == sai_serialize_object_id(br_port_id))
+        {
+            for (auto& attr: bp.second)
+            {
+                 port_id = attr.second->getAttr()->value.oid;
+                 if (port_id)
+                 {
+                     port_id_found = true;
+	             std::string if_name;
+	             bool found = getTapNameFromPortId(port_id, if_name);
+	             if (found == true)
+                     {
+	                 hwifname = tap_to_hwif_name(if_name.c_str());
+	             }
+                }
+            }
+        }
+    }
+    if (port_id_found == false || hwifname == NULL) {
+        SWSS_LOG_NOTICE("No ports found for bridge port id :%s",sai_serialize_object_id(br_port_id).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    auto attr_vlan_member = sai_metadata_get_attr_by_id(SAI_VLAN_MEMBER_ATTR_VLAN_ID, attr_count, attr_list);
+
+    sai_object_id_t vlan_oid;
+
+    if (attr_vlan_member == NULL)
+    {
+	SWSS_LOG_NOTICE("attr SAI_VLAN_MEMBER_ATTR_VLAN_ID was not passed");
+	return SAI_STATUS_FAILURE;
+    } else {
+	vlan_oid = attr_vlan_member->value.oid;
+    }
+    auto attr_vlanid_map = m_objectHash.at(SAI_OBJECT_TYPE_VLAN).at(sai_serialize_object_id(vlan_oid));
+    auto md_vlan_id = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_VLAN, SAI_VLAN_ATTR_VLAN_ID);
+    auto vlan_id =  attr_vlanid_map.at(md_vlan_id->attridname)->getAttr()->value.u16;
+
+    if (vlan_id == 0)
+    {
+        SWSS_LOG_NOTICE("attr VLAN object id  was not passed");
+        return SAI_STATUS_FAILURE;
+    }
+
+    uint32_t bridge_id = (uint32_t)vlan_id;
+    auto attr_tag_mode = sai_metadata_get_attr_by_id(SAI_VLAN_MEMBER_ATTR_VLAN_TAGGING_MODE, attr_count, attr_list);
+    uint32_t tagging_mode = 0;
+    const char *hw_ifname;
+    char host_subifname[32];
+
+    if (attr_tag_mode == NULL)
+    {
+        SWSS_LOG_ERROR("attr SAI_VLAN_MEMBER_ATTR_VLAN_ID was not passed");
+        return SAI_STATUS_FAILURE;
+    }
+    tagging_mode = attr_tag_mode->value.u32;
+    if (tagging_mode != SAI_VLAN_TAGGING_MODE_UNTAGGED)
+    {
+        /*
+         create vpp subinterface and set it as bridge port
+        */
+        snprintf(host_subifname, sizeof(host_subifname), "%s.%u", hwifname, vlan_id);
+
+        /* The host(tap) subinterface is also created as part of the vpp subinterface creation */
+        create_sub_interface(hwifname, vlan_id, vlan_id);
+
+        /* Get new list of physical interfaces from VPP */
+        refresh_interfaces_list();
+
+        hw_ifname = host_subifname;
+
+        //Create bridge and set the l2 port
+        set_sw_interface_l2_bridge(hw_ifname,bridge_id, true);
+
+        //Set interface state up
+        interface_set_state(hw_ifname, true);
+    }else {
```

Comment:
It is better to explicitly test TAGGED and UNTAGEED because there is a third possible option named SAI_VLAN_TAGGING_MODE_PRIORITY_TAGGED.

Reply:
addressed

Reply:
addressed